syntax = "proto3";

package RenderProto;

message Alpha {
    double val = 1;
}

message ColorName {
    string color_name = 1;
}

message Color {
    uint32 red = 1;
    uint32 green = 2;
    uint32 blue = 3;

    Alpha alpha = 4;
    ColorName cn = 5;
}

message RenderSettings {
    double width = 1;
    double height = 2;
    double outer_margin = 3;
    double padding = 4;
    double stop_radius = 5;
    double line_width = 6;
    double company_radius = 7;
    double company_line_width = 8;

    int32 stop_label_font_size = 9;
    repeated double stop_label_offset = 10;

    int32 bus_label_font_size = 11;
    repeated double bus_label_offset = 12;

    Color color = 13;
    double underlayer_width = 14;

    repeated Color color_palette = 15;
    repeated string layers = 16;
}